### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by peterchappy**
@nateabele 100% works, take a pass and let me know where you want refactored more. I'll take another pass on Friday too.

### CODE REVIEW

File: src/components/header/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+type HeaderProps = {
+  addTodo?: () => any;
+}
+
+const placeholder = 'What needs to be done?';
+
+export default ({ addTodo }: HeaderProps) => (
+  <header className='header'>
+    <h1>todos</h1>
+    <TodoTextInput isNew placeholder={placeholder} onSubmit={addTodo} />
```
Comment (nateabele) [inline src/components/header/index.tsx:13]: Is the `isNew` bit valid?

File: src/containers/app/container.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  public static expects = { value: is(Number) };
+};
+
+const createTodoId = () => new Date().valueOf();
+
+const emptyInput = {
+  text: '',
+};
+
+export default container({
```
Comment (nateabele) [inline src/containers/app/container.tsx]: You should be able to do `export default container<TodoAppModel>({` here and then not have to re-type the model in your updaters.

File: src/containers/app/container.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    })],
+
+    [ShowFilter, (state: TodoAppModel, { value }) => mergeDeepRight(state, { filter: value })],
+  ],
+
+  view: ({ emit, todos, filter }) => (
+    <App
+      todos={todos}
+      addTodo={emit(AddTodo)}
+      actions={{
```
Comment (nateabele) [inline src/containers/app/container.tsx]: Get rid of this `actions` guy.

File: src/containers/todo_text_input/index.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  editing?: boolean;
+  isNew?: boolean;
+  placeholder?: string;
+  value?: string;
+  onBlur?: () => any;
+  onChange?: () => any;
+  onSubmit?: () => any;
+}
+
+export default ({ editing = false, isNew = false, placeholder, value = '', onBlur, onChange, onSubmit }: TodoInputProp) => (
```
Comment (nateabele) [inline src/containers/todo_text_input/index.tsx]: You can probably replace the last 5 with `...props` and just pass them through, then use `withProps` to set the default value for `value` if you really need it.

**Review by nateabele — APPROVED** [review]
Overall it looks great, just a couple of non-critical things.

File: src/components/header/index.tsx (hunk 1 above)
Comment (peterchappy) [reply src/components/header/index.tsx:13]: the `isNew` is a class to make the top one look different

**Pull request merged**